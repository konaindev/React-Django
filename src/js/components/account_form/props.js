export const props = {
  steps: [
    { name: "Set Password", isComplete: true },
    { name: "Complete Account", isActive: true }
  ],
  title: "Complete your account",
  subtitle:
    "We need just a bit more information about you to complete your account."
};
