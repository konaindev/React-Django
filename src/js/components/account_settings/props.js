const user = {
  account_id: 1,
  account_name: "Remarkably",
  account_settings_url: "/users/account-settings",
  email: "test@psl.com",
  logout_url: "/users/logout/",
  user_id: "usr_jjzpeyfeshzpaha5"
};

const itemsOrder = ["profile", "lock", "email"];

const rules = [
  {
    label: "Be at least 8 characters",
    key: "length"
  },
  {
    label: "Contain alphabetic characters",
    key: "characters"
  },
  {
    label: "Not match personal information",
    key: "personal"
  },
  {
    label: "Not be a commonly used password",
    key: "used"
  }
];

const company_roles = [
  { label: "Owner", value: "owner" },
  { label: "Developer", value: "developer" },
  { label: "Asset Manager", value: "asset_manager" },
  { label: "Property Manager", value: "property_manager" }
];

const office_options = [
  { label: "Global", value: 1 },
  { label: "National", value: 2 },
  { label: "Regional", value: 3 },
  { label: "Other", value: 4 }
];

const office_countries = [
  { label: "United States of America", value: "USA" },
  { label: "United Kingdom", value: "GBR" }
];

const us_state_list = [
  { label: "Washington", value: "Washington" },
  { label: "New York", value: "New York" }
];

const gb_county_list = [
  { label: "London", value: "London" },
  { label: "Aberdeen", value: "Aberdeen" }
];

const profile = {
  avatar_url:
    "https://lh3.googleusercontent.com/-cQLcFi7r2uc/AAAAAAAAAAI/AAAAAAAAAAA/ACHi3rfoCSVbR8qVruV55uAYdSC-znVn2w.CMID/s96-c/photo.jpg",
  first_name: "Phillip",
  last_name: "McPhillipson",
  title: "Founder",
  phone_country_code: "",
  phone: "",
  phone_ext: "",
  company: { label: "Glacier Associates", value: "123456789" },
  company_roles: ["owner", "asset_manager", "property_manager"],
  office_street: "1730 Minor Avenue",
  office_city: "Lansing",
  office_state: { label: "Michigan", value: "Michigan" },
  office_zip: "11111",
  office_country: { label: "United States of America", value: "USA" },
  office_address: "1730 Minor Avenue, Lansing, MI",
  office_name: "Michigan",
  office_type: 3
};

const tabsOrder = ["portfolio", "group", "property"];

const initialTab = "portfolio";

export const props = {
  rules,
  profile,
  user,
  company_roles,
  office_options,
  itemsOrder,
  tabsOrder,
  initialTab,
  office_countries,
  us_state_list,
  gb_county_list,
  dispatch: () => {}
};
