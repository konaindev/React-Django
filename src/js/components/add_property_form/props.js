export const props = {
  post_url: "/projects/new",
  states: [
    { id: "AL", name: "AL" },
    { id: "GA", name: "GA" },
    { id: "NE", name: "NE" },
    { id: "OR", name: "OR" },
    { id: "WY", name: "WY" }
  ],
  packages: [
    { id: "accelerate", name: "Accelerate" },
    { id: "optimize", name: "Optimize" },
    { id: "ground", name: "Ground Up" },
    { id: "other", name: "Not Sure" }
  ]
};
