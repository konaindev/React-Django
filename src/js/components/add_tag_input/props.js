const suggestedTags = [
  { name: "Tag 1", count: 5 },
  { name: "Test Tag 2", count: 3 },
  { name: "Lake Highland Properties 3", count: 2 },
  { name: "New Tag", count: 1 }
];

export default { suggestedTags };
