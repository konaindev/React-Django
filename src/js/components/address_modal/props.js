export default {
  isOpen: true,
  addresses: {
    suggested_address: {
      office_city: "Washington",
      office_state: "DC",
      office_street: "1600 Pennsylvania Ave NW",
      office_zip: "20500",
      office_country: "United States"
    }
  },
  dispatch_type: "FAKE DISPATCH"
};
