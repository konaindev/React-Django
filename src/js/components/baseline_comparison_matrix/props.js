import {
  baselineReport,
  baselineReportWithoutCompetitors,
  baselineReportWithOneCompetitor
} from "../common_report/props";

export const props = { report: baselineReport };
export const no_competitor_props = { report: baselineReportWithoutCompetitors };
export const one_competitor_props = {
  report: baselineReportWithOneCompetitor
};
