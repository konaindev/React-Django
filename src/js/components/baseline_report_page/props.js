import { project, report_links } from "../project_page/props";
import { funnel_history } from "../funnel_performance_analysis/FunnelProps";

export const report = {
  dates: {
    start: "2017-07-24",
    end: "2018-07-23"
  },
  property_name: "Portland Multi-Family",
  property: {
    average_monthly_rent: "1847.00",
    lowest_monthly_rent: "1847.00",
    cost_per_exe_vs_rent: 0.54,
    total_units: 201,
    leasing: {
      change: 36,
      cds: 28,
      cd_rate: 0.29,
      renewal_notices: 94,
      renewals: 94,
      renewal_rate: 0.71,
      resident_decisions: 132,
      vacation_notices: 38,
      rate: 0.74,
      units: 192
    },
    occupancy: {
      move_ins: 72,
      move_outs: 36,
      rate: 0.71,
      units: 185,
      occupiable: 260
    }
  },
  funnel: {
    volumes: {
      usv: 19621,
      inq: 785,
      tou: 259,
      app: 96,
      exe: 68
    },
    costs: {
      usv: "3.47",
      inq: "86.62",
      tou: "262.00",
      app: "708.00",
      exe: "1000.00"
    },
    conversions: {
      usv_inq: 0.04,
      inq_tou: 0.33,
      tou_app: 0.37,
      app_exe: 0.71,
      usv_exe: 0.003
    }
  },
  investment: {
    acquisition: {
      expenses: {
        demand_creation: "48000.00",
        leasing_enablement: "0.00",
        market_intelligence: "0.00",
        reputation_building: "20000.00"
      },
      total: "68000.00",
      romi: 10,
      estimated_revenue_gain: "709200.00"
    },
    retention: {
      expenses: {
        demand_creation: "0.00",
        leasing_enablement: "6000.00",
        market_intelligence: "0.00",
        reputation_building: "0.00"
      },
      total: "6000.00",
      romi: 346,
      estimated_revenue_gain: "2080000.00"
    },
    total: {
      total: "74000.00",
      romi: 38,
      estimated_revenue_gain: "2800000.00"
    }
  },
  four_week_funnel_averages: {
    usv: 1509,
    inq: 60,
    tou: 20,
    app: 7,
    exe: 5
  },
  funnel_history
};

const current_report_link = report_links.baseline;

const share_info = {
  shared: true,
  share_url: `/projects/${project.public_id}/share/baseline/`,
  update_action: "shared_reports"
};

export default {
  project,
  report,
  report_links,
  current_report_link,
  share_info
};
