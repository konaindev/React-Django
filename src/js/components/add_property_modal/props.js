import { states } from "../add_property_form/states";

export const props = {
  formProps: {
    post_url: null,
    packages: [
      { id: "accelerate", name: "Accelerate" },
      { id: "optimize", name: "Optimize" },
      { id: "ground", name: "Ground Up" },
      { id: "other", name: "Not Sure" }
    ],
    states
  }
};
