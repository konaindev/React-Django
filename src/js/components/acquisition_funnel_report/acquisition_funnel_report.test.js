import { shallow } from "enzyme";

import AcquisitionFunnelReport from "./index";

import { props_baseline, props_performance } from "./props";

describe("AcquisitionFunnelReport", () => {
  it("renders baseline report correctly", () => {
    let tree = shallow(<AcquisitionFunnelReport {...props_baseline} />);
    expect(tree.debug()).toMatchSnapshot();

    tree = shallow(<AcquisitionFunnelReport.HeadlineNumbers {...props_baseline} />);
    expect(tree.debug()).toMatchSnapshot();

    tree = shallow(<AcquisitionFunnelReport.FunnelContent {...props_baseline} />);
    expect(tree.debug()).toMatchSnapshot();
  });

  it("renders performance report correctly", () => {
    let tree = shallow(<AcquisitionFunnelReport {...props_performance} />);
    expect(tree.debug()).toMatchSnapshot();

    tree = shallow(<AcquisitionFunnelReport.HeadlineNumbers {...props_performance} />);
    expect(tree.debug()).toMatchSnapshot();

    tree = shallow(<AcquisitionFunnelReport.FunnelContent {...props_performance} />);
    expect(tree.debug()).toMatchSnapshot();
  });
});
