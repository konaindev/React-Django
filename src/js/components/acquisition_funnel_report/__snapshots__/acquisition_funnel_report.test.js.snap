// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcquisitionFunnelReport renders baseline report correctly 1`] = `
"<ReportSection name=\\"Acquisition Funnel\\">
  <Component report={{...}} />
  <Component>
    <Component report={{...}} type=\\"baseline\\" />
  </Component>
</ReportSection>"
`;

exports[`AcquisitionFunnelReport renders baseline report correctly 2`] = `
"<BoxRow spacing=\\"normal\\">
  <PercentageGraphBox name=\\"USV > EXE\\" infoTooltip=\\"usv_to_exe\\" value={0.003} target={[undefined]} delta={[undefined]} series={[undefined]} digits={2} />
  <PercentageGraphBox name=\\"Cancellation & Denial Rate\\" value={0.29} target={[undefined]} delta={[undefined]} series={[undefined]} />
  <PercentageGraphBox name=\\"Cost per Exe / Lowest Monthly Rent\\" infoTooltip=\\"cost_per_exe_lowest_monthly_rent\\" value={0.54} target={[undefined]} delta={[undefined]} series={[undefined]} extraContent=\\"$1,000 / $1,847\\" />
</BoxRow>"
`;

exports[`AcquisitionFunnelReport renders baseline report correctly 3`] = `
"<Component>
  <Component>
    <Component name=\\"Volume of Activity\\" />
    <Component>
      <FunnelNumberBox name=\\"Volume of USV\\" infoTooltip=\\"volume_of_usv\\" value={19621} target={2136} delta={[undefined]} targetFormatter={[Function: format4WeekAverage]} />
      <FunnelNumberBox name=\\"Volume of INQ\\" infoTooltip=\\"volume_of_inq\\" value={785} target={128} delta={[undefined]} targetFormatter={[Function: format4WeekAverage]} />
      <FunnelNumberBox name=\\"Volume of TOU\\" infoTooltip=\\"volume_of_tou\\" value={259} target={52} delta={[undefined]} targetFormatter={[Function: format4WeekAverage]} />
      <FunnelNumberBox name=\\"Volume of APP\\" infoTooltip=\\"volume_of_app\\" value={96} target={19} delta={[undefined]} targetFormatter={[Function: format4WeekAverage]} />
      <FunnelNumberBox name=\\"Volume of EXE\\" infoTooltip=\\"volume_of_exe\\" value={68} target={16} delta={[undefined]} targetFormatter={[Function: format4WeekAverage]} />
    </Component>
  </Component>
  <Component>
    <Component name=\\"Conversion Rate\\" />
    <Component>
      <FunnelPercentBox name=\\"USV → INQ\\" infoTooltip=\\"usv_to_inq\\" value={0.04} target={[undefined]} delta={[undefined]} />
      <FunnelPercentBox name=\\"INQ → TOU\\" infoTooltip=\\"inq_to_tou\\" value={0.33} target={[undefined]} delta={[undefined]} />
      <FunnelPercentBox name=\\"TOU → APP\\" infoTooltip=\\"tou_to_app\\" value={0.37} target={[undefined]} delta={[undefined]} />
      <FunnelPercentBox name=\\"APP → EXE\\" infoTooltip=\\"app_to_exe\\" value={0.71} target={[undefined]} delta={[undefined]} />
    </Component>
  </Component>
  <Component>
    <Component name=\\"Cost per Activity\\" />
    <Component>
      <FunnelCurrencyBox name=\\"Cost per USV\\" infoTooltip=\\"cost_per_usv\\" value=\\"3.47\\" target={[undefined]} delta={[undefined]} />
      <FunnelCurrencyBox name=\\"Cost per INQ\\" infoTooltip=\\"cost_per_inq\\" value=\\"86.62\\" target={[undefined]} delta={[undefined]} />
      <FunnelCurrencyBox name=\\"Cost per TOU\\" infoTooltip=\\"cost_per_tou\\" value=\\"262.00\\" target={[undefined]} delta={[undefined]} />
      <FunnelCurrencyBox name=\\"Cost per APP\\" infoTooltip=\\"cost_per_app\\" value=\\"708.00\\" target={[undefined]} delta={[undefined]} />
      <FunnelCurrencyBox name=\\"Cost per EXE\\" infoTooltip=\\"cost_per_exe\\" value=\\"1000.00\\" target={[undefined]} delta={[undefined]} />
    </Component>
  </Component>
</Component>"
`;

exports[`AcquisitionFunnelReport renders performance report correctly 1`] = `
"<ReportSection name=\\"Acquisition Funnel\\">
  <Component report={{...}} />
  <Component>
    <Component report={{...}} type=\\"performance\\" />
  </Component>
</ReportSection>"
`;

exports[`AcquisitionFunnelReport renders performance report correctly 2`] = `
"<BoxRow spacing=\\"normal\\">
  <PercentageGraphBox name=\\"USV > EXE\\" infoTooltip=\\"usv_to_exe\\" value={0.004} target={0.007} delta={-0.001} series={{...}} digits={2} />
  <PercentageGraphBox name=\\"Cancellation & Denial Rate\\" value={0.2} target={0.2} delta={0.03} series={{...}} />
  <PercentageGraphBox name=\\"Cost per Exe / Lowest Monthly Rent\\" infoTooltip=\\"cost_per_exe_lowest_monthly_rent\\" value={0.58} target={1.03} delta={-0.2} series={{...}} extraContent=\\"$1,077 / $1,857\\" />
</BoxRow>"
`;

exports[`AcquisitionFunnelReport renders performance report correctly 3`] = `
"<Component>
  <Component>
    <Component name=\\"Volume of Activity\\" />
    <Component>
      <FunnelNumberBox name=\\"Volume of USV\\" infoTooltip=\\"volume_of_usv\\" value={3008} target={2136} delta={423} targetFormatter={[Function]} />
      <FunnelNumberBox name=\\"Volume of INQ\\" infoTooltip=\\"volume_of_inq\\" value={150} target={128} delta={-19} targetFormatter={[Function]} />
      <FunnelNumberBox name=\\"Volume of TOU\\" infoTooltip=\\"volume_of_tou\\" value={57} target={52} delta={2} targetFormatter={[Function]} />
      <FunnelNumberBox name=\\"Volume of APP\\" infoTooltip=\\"volume_of_app\\" value={16} target={19} delta={-2} targetFormatter={[Function]} />
      <FunnelNumberBox name=\\"Volume of EXE\\" infoTooltip=\\"volume_of_exe\\" value={13} target={16} delta={-1} targetFormatter={[Function]} />
    </Component>
  </Component>
  <Component>
    <Component name=\\"Conversion Rate\\" />
    <Component>
      <FunnelPercentBox name=\\"USV → INQ\\" infoTooltip=\\"usv_to_inq\\" value={0.05} target={0.06} delta={-0.01} />
      <FunnelPercentBox name=\\"INQ → TOU\\" infoTooltip=\\"inq_to_tou\\" value={0.38} target={0.4} delta={0.01} />
      <FunnelPercentBox name=\\"TOU → APP\\" infoTooltip=\\"tou_to_app\\" value={0.28} target={0.36} delta={-0.02} />
      <FunnelPercentBox name=\\"APP → EXE\\" infoTooltip=\\"app_to_exe\\" value={0.8} target={0.8} delta={0.02} />
    </Component>
  </Component>
  <Component>
    <Component name=\\"Cost per Activity\\" />
    <Component>
      <FunnelCurrencyBox name=\\"Cost per USV\\" infoTooltip=\\"cost_per_usv\\" value=\\"4.32\\" target=\\"9.90\\" delta=\\"1.67\\" />
      <FunnelCurrencyBox name=\\"Cost per INQ\\" infoTooltip=\\"cost_per_inq\\" value=\\"86.67\\" target=\\"165.00\\" delta=\\"24.76\\" />
      <FunnelCurrencyBox name=\\"Cost per TOU\\" infoTooltip=\\"cost_per_tou\\" value=\\"246.00\\" target=\\"413.00\\" delta=\\"-120.00\\" />
      <FunnelCurrencyBox name=\\"Cost per APP\\" infoTooltip=\\"cost_per_app\\" value=\\"875.00\\" target=\\"1515.00\\" delta=\\"-202.00\\" />
      <FunnelCurrencyBox name=\\"Cost per EXE\\" infoTooltip=\\"cost_per_exe\\" value=\\"1077.44\\" target=\\"1909.00\\" delta=\\"-539.00\\" />
    </Component>
  </Component>
</Component>"
`;
